.github/workflows/gpu-tests.yaml

```diff
@@ -0,0 +1,60 @@
+# This workflow runs all of the very-long tests within main.py
+
+name: Weekly Tests
+
+on:
+  # Runs every Sunday from 7AM UTC
+  schedule:
+    - cron:  '00 7 * * 6'
+  # Allows us to manually start workflow for testing
+  workflow_dispatch:
+
+jobs:
+  build-gem5:
+    runs-on: [self-hosted, linux, x64, build]
+    container: gcr.io/gem5-test/gcn-gpu:latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Scheduled workflows run on the default branch by default. We
+          # therefore need to explicitly checkout the develop branch.
+          ref: develop
+      - name: Build gem5
+        run: scons build/GCN3_X86/gem5.opt -j $(nproc) --ignore-style
+      - uses: actions/upload-artifact@v3
+        with:
+          name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-gcn3
+          path: build/GCN3_X86/gem5.opt
+          retention-days: 5
+      - run: echo "This job's status is ${{ job.status }}."
+
+  HACC-tests:
+    runs-on: [self-hosted, linux, x64, build]
+    container: gcr.io/gem5-test/gcn-gpu:latest
+    needs: build-gem5
+    timeout-minutes: 4320 # 3 day
```

Comment:
Honestly if HACC takes more than a day I would be surprised -- are you seeing it take longer?  I'm fine with leaving this if it is convenient, but it does seem like a very conservative boundary.

Reply:
I agree here, I think this test actually only takes a few minutes with the given input size, so anything longer than maybe a few hours is indicative of a hang

Reply:
Ah I can update this, this value was carried over from when we were setting the timeout for all the weekly tests to be run together and was based on a conservative estimate from our weekly.sh tests, but I can decrease it since it does finish quickly